packages/tao/src/shared/nx-plugin.ts

```diff
@@ -25,30 +33,45 @@ export interface NxPlugin {
   projectFilePatterns?: string[];
 }
 
-function findPluginPackageJson(path: string, plugin: string) {
-  while (true) {
-    if (!path.startsWith(appRootPath)) {
-      throw new Error("Couldn't find a package.json for Nx plugin:" + plugin);
-    }
-    if (existsSync(join(path, 'package.json'))) {
-      return join(path, 'package.json');
-    }
-    path = dirname(path);
-  }
-}
-
+// Short lived cache (cleared between cmd runs)
+// holding resolved nx plugin objects.
+// Allows loadNxPlugins to be called multiple times w/o
+// executing resolution mulitple times.
 let nxPluginCache: NxPlugin[] = null;
-export function loadNxPlugins(plugins?: string[]): NxPlugin[] {
+export function loadNxPlugins(
+  plugins?: string[],
+  paths = [appRootPath]
+): NxPlugin[] {
   return plugins?.length
     ? nxPluginCache ||
-        (nxPluginCache = plugins.map((path) => {
-          const pluginPath = require.resolve(path, {
-            paths: [appRootPath],
-          });
-
-          const { name } = readJsonFile(
-            findPluginPackageJson(pluginPath, path)
-          );
+        (nxPluginCache = plugins.map((moduleName) => {
+          let pluginPath: string;
+          try {
+            pluginPath = require.resolve(moduleName, {
+              paths,
+            });
+          } catch (e) {
+            if (e.code === 'MODULE_NOT_FOUND') {
+              const plugin = resolveLocalNxPlugin(moduleName);
+              const { main } = Object.values(plugin.projectConfig.targets).find(
```

Comment:
Also, it's `@nrwl/node:webpack` now, not `package`.

Reply:
Package is actually removed in favor of tsc/swc,  it never bundled. I included it here for legacy reasons, but it could be removed honestly.

Reply:
> Wouldn't the destructure fail if there are no options?

Yeah, it would. For anything generated with `nx-plugin` that shouldn't be the case, but it may be worth adding some fallbacks here in case someone is using a custom executor for their plugin.